src/components/TournamentMarker.jsx

```diff
@@ -0,0 +1,53 @@
+import {useEffect, useRef, memo, useContext, useCallback} from 'react';
+
+import L from 'leaflet';
+import * as ReactLeaflet from 'react-leaflet';
+
+import icon from "leaflet/dist/images/marker-icon.png";
+import icon2x from "leaflet/dist/images/marker-icon-2x.png";
+import iconShadow from "leaflet/dist/images/marker-shadow.png";
+
+import VisibleTournamentsTrackingContext from '../contexts/VisibleTournamentsTrackingContext';
+
+let defaultIcon = new L.Icon.Default();
+
+defaultIcon.options.iconUrl = icon;
+defaultIcon.options.iconRetinaUrl = icon2x;
+defaultIcon.options.shadowUrl = iconShadow;
+
+const TournamentMarker = memo(({ tournament }) => {
+    const markerRef = useRef(null);
+
+    const { watch, unwatch } = useContext(VisibleTournamentsTrackingContext);
+
+    const openPopup = useCallback(() => {
+        markerRef.current?.togglePopup();
+    }, []);
+
+    useEffect(() => {
+        const element = markerRef.current?.getElement();
+
```

Comment:
nit:  consider removing the newline, since the if check is associated with this assignment?

Reply:
Done.